src/utils/Firebase/firebase.js

```diff
@@ -226,40 +242,71 @@ class Firebase {
         let batch = this.db.batch();
 
         // Add/update deploymentAccount references to this custom lesson
-        for (const deploymentAccountId of deploymentAccountIds) {
-          let deploymentRef = this.db.doc(
-            `deployment_account/${deploymentAccountId}/`
-          );
-          batch.update(deploymentRef, {
-            customLessons: app.firestore.FieldValue.arrayUnion(
-              customLessonRef.id
-            )
-          });
-        }
+        let deploymentRefs = deploymentAccountIds.map(id =>
+          this.db.doc(`deployment_account/${id}/`)
+        );
+
+        return Promise.all(deploymentRefs.map(ref => ref.get()))
+          .then(deploymentAccountDocs => {
+            for (let i = 0; i < deploymentRefs.length; i++) {
+              let deploymentRef = deploymentRefs[i];
```

Comment:
this for loop is far too big what is happening in here

Reply:
and what's the reason that you didn't do a `.foreach` or something

Reply:
Yeah this was really big imo lol but I had to track it with the variable i to make sure i was accessing the same reference/id from the lists, which aren't consistent with .forEach

Reply:
makes sense